File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int area = brown+yellow;
    for(int height=3;;height++){
        if(area%height!=0) continue;
        
        int width = area/height;
        int yNum = (width-2)*(height-2);
        
        if(yNum==yellow){
            answer.push_back(width);
            answer.push_back(height);
            break;
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.cs
using System;

public class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
         for(int i=1;i<=yellow;i++){
             if(yellow%i!=0) continue;
             int a = i;
             int b = yellow/a;
             
             int sum = a*2+b*2+4;
             
             if(sum==brown){
                 answer[0] = b+2;
                 answer[1] = a+2;
                
                 break;
             }
         }
        return answer;
    }
}